### PR TITLE
fritz-exporter: init at 2.3.1

### LINKED ISSUE #277828: Package request: fritz_exporter
**Project description**

Prometheus Exporter for AVM FritzBox

**Metadata**

* homepage URL: https://github.com/pdreker/fritz_exporter
* source URL: https://github.com/pdreker/fritz_exporter
* license: AL2
* platforms: linux, probably also darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
fixes https://github.com/NixOS/nixpkgs/issues/277828

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fr/fritz-exporter/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    pytestCheckHook
+  ];
+
+  meta = {
+    changelog = "https://github.com/pdreker/fritz_exporter/blob/${src.rev}/CHANGELOG.md";
+    description = "Prometheus exporter for Fritz!Box home routers";
+    homepage = "https://github.com/pdreker/fritz_exporter";
+    license = lib.licenses.asl20;
+    mainProgram = "fritzexporter";
+    maintainers = with lib.maintainers; [ ];
```
Comment (dotlambda) [inline pkgs/by-name/fr/fritz-exporter/package.nix:50]: @NyCodeGHG?

Comment (NyCodeGHG) [reply pkgs/by-name/fr/fritz-exporter/package.nix:50]: yes, please add me as a maintainer, my maintainer name is `marie`
i'll try to test this package tomorrow

**Comment by Shawn8901** [comment]
Added the package to my personal setup. Works fine so far, tried to scrap different values (connected wifi devices, dsl datarate and smart home temp).
At some point it might be useful to add a module for that, possibly it make sense to add it in the prometheus scope, like the other exporters.
A fast hacked module that i used for testing can be found here https://github.com/Shawn8901/nix-configuration/commit/1c64c6e1e4f539f04730d9928edb2d6b20740dac

**Review by Shawn8901 — APPROVED** [review]

**Comment by NyCodeGHG** [comment]
Quick test with my fritzbox, looks fine

**Review by NyCodeGHG — APPROVED** [review]

**Pull request merged**